modules/seqexec/server/src/main/scala/seqexec/server/SeqTranslate.scala

```diff
@@ -320,20 +317,47 @@ object SeqTranslate {
 
     def toInstrumentSys(inst: Instrument)(
       implicit ev: Timer[F], cio: Concurrent[F]
-    ): F[InstrumentSystem[F]] = inst match {
-      case Instrument.F2    => Flamingos2(systems.flamingos2, systems.dhs).pure[F].widen[InstrumentSystem[F]]
-      case Instrument.GmosS => GmosSouth(systems.gmosSouth, systems.dhs, gmosNsCmd).pure[F].widen[InstrumentSystem[F]]
-      case Instrument.GmosN => GmosNorth(systems.gmosNorth, systems.dhs, gmosNsCmd).pure[F].widen[InstrumentSystem[F]]
-      case Instrument.Gnirs => Gnirs(systems.gnirs, systems.dhs).pure[F].widen[InstrumentSystem[F]]
-      case Instrument.Gpi   => Gpi(systems.gpi).pure[F].widen[InstrumentSystem[F]]
-      case Instrument.Ghost => Ghost(systems.ghost).pure[F].widen[InstrumentSystem[F]]
-      case Instrument.Niri  => Niri(systems.niri, systems.dhs).pure[F].widen[InstrumentSystem[F]]
-      case Instrument.Nifs  => Nifs(systems.nifs, systems.dhs).pure[F].widen[InstrumentSystem[F]]
-      case Instrument.Gsaoi => Gsaoi(systems.gsaoi, systems.dhs).pure[F].widen[InstrumentSystem[F]]
-      case _                => Unexpected(s"Instrument $inst not supported.").raiseError[F, InstrumentSystem[F]]
+    ): F[SystemOverrides => InstrumentSystem[F]] = inst match {
+      case Instrument.F2    => { ov:SystemOverrides =>
+        Flamingos2(overriddenSystems.flamingos2(ov), overriddenSystems.dhs(ov)):InstrumentSystem[F]
+      }.pure[F]
+      case Instrument.GmosS => { ov:SystemOverrides =>
+        GmosSouth(overriddenSystems.gmosSouth(ov), overriddenSystems.dhs(ov), gmosNsCmd):InstrumentSystem[F]
+      }.pure[F]
+      case Instrument.GmosN => { ov:SystemOverrides =>
+        GmosNorth(overriddenSystems.gmosNorth(ov), overriddenSystems.dhs(ov), gmosNsCmd):InstrumentSystem[F]
+      }.pure[F]
+      case Instrument.Gnirs => { ov:SystemOverrides =>
+        Gnirs(overriddenSystems.gnirs(ov), overriddenSystems.dhs(ov)):InstrumentSystem[F]
+      }.pure[F]
+      case Instrument.Gpi   => { ov:SystemOverrides =>
+        Gpi(overriddenSystems.gpi(ov)):InstrumentSystem[F]
+      }.pure[F]
+      case Instrument.Ghost => { ov:SystemOverrides =>
+        Ghost(overriddenSystems.ghost(ov)):InstrumentSystem[F]
+      }.pure[F]
+      case Instrument.Niri  => { ov:SystemOverrides =>
+        Niri(overriddenSystems.niri(ov), overriddenSystems.dhs(ov)):InstrumentSystem[F]
+      }.pure[F]
+      case Instrument.Nifs  => { ov:SystemOverrides =>
+        Nifs(overriddenSystems.nifs(ov), overriddenSystems.dhs(ov)):InstrumentSystem[F]
+      }.pure[F]
+      case Instrument.Gsaoi => { ov:SystemOverrides =>
+        Gsaoi(overriddenSystems.gsaoi(ov), overriddenSystems.dhs(ov)):InstrumentSystem[F]
+      }.pure[F]
     }
 
-    private def calcResources(sys: List[System[F]]): Set[Resource] = sys.map(_.resource).toSet
+    def instrumentSpecs(instrument: Instrument): InstrumentSpecifics = instrument match {
```

Comment:
Could we have an `Instrument{ def specifics: InstrumentSpecifics}` method?

Reply:
It would be handy, but Instrument and InstrumentSpecifics live in separate packages with separate functions. It also would invert the current dependency between those packages.